back/app/models/custom_page.rb

```diff
@@ -46,16 +43,15 @@ class CustomPage < ApplicationRecord
 
   accepts_nested_attributes_for :pinned_admin_publications, allow_destroy: true
 
+  before_validation :strip_title
+  before_validation :generate_slug, on: :create
```

Comment:
Both of these are in `StaticPage` model and seems wise to include them here, especially `:generate_slug`

Reply:
Agreed. :+1: